### DISCUSSION TITLE
Empty query for `workspace/symbol` should return all symbols in the workspace

### DISCUSSION THREAD

**Discussion by MatejKastak** [microsoft/pyright#9305] [Ideas]
Hello, thank you for the pyright.

Based on the spec ([WorkspaceSymbolParams](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspaceSymbolParams)) for query parameter:
> A query string to filter symbols by. **Clients may send an empty string here to request all symbols.**

But the [implementation](https://github.com/microsoft/pyright/blob/main/packages/pyright-internal/src/languageService/workspaceSymbolProvider.ts#L131-L135) is directly ignoring the spec.
> // Don't do a search if the query is empty. We'll return too many results in this case.

### Question
What should I do in this case if I want the full list, and I acknowledge that it might be too big?

Would it be possible to return the full list for the empty query, for users that opt-in (for example via config)? So that you can at least opt-in to spec compliant behavior?

### My motivation:
When searching for a symbol in the workspace, I would like to use filtering (fuzzy matching) from my editor, instead of the non-fuzzy matching of the query (which causes inconvenience when searching for symbols).

**Comment by erictraut** [comment]
Pyright is focused on type checking. It has some basic language server support that is maintained by the Microsoft Pylance team. If you would like to suggest changes or enhancements to this functionality, please file an issue in the [pylance-release](https://github.com/microsoft/pylance-release/issues) issue tracker or discussion section.

**Reply by MatejKastak** [reply]
Thank you for the suggestion, I went over the repo that you linked and found a similar issue. I followed up this discussion in there https://github.com/microsoft/pylance-release/discussions/5556.
